cmd/main.go

```diff
@@ -205,17 +218,34 @@ func postRun(cmd *cobra.Command, args []string) {
 		if len(reportPathVar) > 0 {
 			err := report.WriteFile(reportPathVar, cfg)
 			if err != nil {
-				log.Error().Msgf("Failed to create report file with error: %s", err)
+				return fmt.Errorf("failed to create report file with error: %s", err)
 			}
 		}
 	} else {
-		log.Error().Msg("Scan completed with empty content")
-		os.Exit(0)
+		fmt.Println("Scan completed with empty content")
+		return nil
```

Comment:
I think it will be better to finish the program always in one place, (on `main`).

To do that I can help you, I'm throwing here an idea from my mind, I didn't check it but I don't want to lose it.

1. Create an error channel.
2. From the go routine, send the error to the channel, or close the channel if no error.
3. After the go routine, read the error channel.
4. If you read an error, return it.
5. If you read `nil`, check the results count.

Reply:
Sounds interesting, will have to do some research in order to better understand the concept and probably come back with some questions / code to discuss.

Reply:
Proposed Changes:

log.fatal error in the preRun function:
- Added error channel
- Replaced log.fatal within the go routine under preRun with redirection of the err to the error channel 
- Added a check in the postRun function to determine whether the channel contains an error

Important Note:

While debugging I noticed that channels.WaitGroup.Wait() within postRun ends before the counter increase / first item is added.

Therefore it is basically skipped. Am I missing something?


TotalSecretsFound is greater than 0:

TotalSecretsFound is an exception case whether we want to exit with non-zero status code but avoid printing an error (as no error occurred).

As far as I could see, if error is returned from post/pre run then it will be printed along with a usage output (regardless of any intervention from our end).

As this is a special we could handle/bypass it by moving the actual "TotalSecretsFound > 0" condition to the Execute function after rootCmd.Execute().

In that case we will return an empty error to the main function. that would lead to a print of an empty line after the scan in case secrets are found.

A different idea is to not consider TotalSecretsFound as an error and add a check in the main function for that situation.


Further thoughts:
As errors are getting printed automatically from post/pre run functions. perhaps we can avoid the error print in the main function.